developing.md

```diff
@@ -0,0 +1,136 @@
+# Developing ember-toucan-core
```

Comment:
View this [rendered](https://github.com/CrowdStrike/ember-toucan-core/blob/ae9d1863380f4c90c798e3eaaf7d501a12cadbe2/developing.md)

Reply:
This is great!

But there is some overlap with `CONTRIBUTING.md`. Have you considered replacing it with this one? Github somehow knows about that file name, so seems better to use that! See what a new contributor gets:

<img width="292" alt="image" src="https://user-images.githubusercontent.com/1325249/224283074-77d1b06f-2711-428d-8523-0ead7c73ca60.png">
(the link goes to `CONTRIBUTING.md`!)

Reply:
Doh! I completely forgot about that haha. I like it! I'll move this to contributing instead